clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -317,32 +326,22 @@ public boolean mkdirs(Path path, FsPermission fsPermission) throws IOException {
     public FileStatus getFileStatus(Path path) throws IOException {
         LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ getFileStatus, path: {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$ ", path.toString());
         ObjectLocation objectLoc = pathToObjectLocation(path);
+        ObjectsApi objectsApi = new ObjectsApi(this.apiClient);
         try {
-            ObjectsApi objectsApi = new ObjectsApi(this.apiClient);
             ObjectStats objectStat = objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath());
-            long length = 0;
-
-
-            Long sizeBytes = objectStat.getSizeBytes();
-            if (sizeBytes != null) {
-                length = sizeBytes;
-            }
-            long modificationTime = 0;
-            Long mtime = objectStat.getMtime();
-            if (mtime != null) {
-                modificationTime = TimeUnit.SECONDS.toMillis(mtime);
-            }
-            Path filePath = path.makeQualified(this.uri, this.workingDirectory);
-            long blockSize = withFileSystemAndTranslatedPhysicalPath(objectStat.getPhysicalAddress(), (FileSystem fs, Path p) -> fs.getDefaultBlockSize(p));
-            return new FileStatus(length, false, 0, blockSize, modificationTime, filePath);
+            return convertObjectStatsToFileStatus(objectStat);
         } catch (ApiException e) {
-            if (e.getCode() == HttpStatus.SC_NOT_FOUND) {
-                throw new FileNotFoundException(path + " not found");
+            if (e.getCode() != HttpStatus.SC_NOT_FOUND) {
+                throw new IOException("statObject", e);
             }
-            throw new IOException("statObject", e);
-        } catch (java.net.URISyntaxException e) {
-            throw new IOException("underlying storage uri", e);
         }
+        // check if path is a directory
```

Comment:
This logic is obviously correct but subtle.  Maybe aid the reader with some more documentation (but open to any further ideas, of course)!
```suggestion
        // not found as a file; check if path is a "directory", i.e. a prefix.
```